go/vt/vttablet/tabletmanager/vreplication/vreplicator.go

```diff
@@ -406,6 +429,38 @@ func (vr *vreplicator) resetFKCheckAfterCopy() error {
 	return err
 }
 
+func (vr *vreplicator) setSQLMode() (func(), error) {
+	// First save the original SQL mode if we have not already done so
+	if vr.originalSQLMode == "" {
+		res, err := vr.dbClient.Execute(getSQLModeQuery)
+		if err != nil || len(res.Rows) != 1 {
+			return nil, fmt.Errorf("could not get the original sql_mode on target: %v", err)
+		}
+		vr.originalSQLMode = res.Rows[0][0].ToString()
+	}
+
+	// Now set it to a permissive mode that will allow us to recreate
+	// any database object that exists on the source in full on the
+	// target
+	query := fmt.Sprintf(setSQLModeQueryf, SQLMode)
+	_, err := vr.dbClient.Execute(query)
+	if err != nil {
+		return nil, fmt.Errorf("could not set the permissive sql_mode on target using %s: %v", query, err)
```

Comment:
(continued from above)
```suggestion
		return resetFunc, fmt.Errorf("could not set the permissive sql_mode on target using %s: %v", query, err)
```